namespace.py

```diff
@@ -351,6 +351,16 @@ def check(self, val):
 
 
 
+class StrOrNone(ValueProcessor):
+  """Allows str, unicode, or None."""
+
+  def check(self, val):
+    if val is not None:
+      Str().check(val)
+
```

Comment:
AFAIU `ValueProcessor` sub-classes do nothing if a check encounters a valid value or raise an exception if the value is invalid.
This new ValueProcessor does nothing if the value is `None` and uses the existing `Str` checker otherwise.